blueprint/core/blueprint_EcmascriptEngine.cpp

```diff
@@ -126,6 +161,20 @@ namespace blueprint
         return result;
     }
 
+    juce::var EcmascriptEngine::evaluate (const juce::File& code)
+    {
+        juce::var result = juce::var::undefined();
+
+        if (detail::safeCompileFile(ctx, code, onUncaughtError))
```

Comment:
Maybe we should return the result in a successful run also? The bundles we're using never really return anything but I could imagine people using `EcmascriptEngine` for all sorts of things.

```
if (detail::safeCompileFile(ctx, code, onUncaughtError))
    duk_call(ctx, 0);

result = readVarFromDukStack(ctx, -1);
duk_pop(ctx);

return result;
```

Reply:
Oh yeah. I think there will only be a result object on the stack in an error state with the duk_compile calls. Otherwise it's a function object. I'll sanity check when I get back to me desk.

Reply:
Ah right but then executing that function should return a result right? So if the compile succeeds, execute whatevers on top and return the result. If the compile fails, return whatevers on top because that shows the error. Although maybe I'm overthinking it... even if your bundle had a return value, it might be the case that when duktape compiles it and puts a function at stack top, _that_ function might not have a return.

/shrug maybe ignore me :)

Reply:
Yeah. Getting back to this shortly so I'll.just run it under the debugger and check. Not super clear from the API docs but I've not diced into duktape itself much yet.

Reply:
@nick-thompson, you are indeed correct. https://duktape.org/api.html#duk_call

Reply:
I miss understood the docs

Reply:
Cool, thanks for the link

Reply:
@nick-thompson, Now that I'm taking a second pass, I think I might change this to use duk_pcall for safety in the same way that you have duk_pcal in EcmascriptEngine::invoke.